core/src/main/scala/org/apache/spark/rdd/ParallelCollectionRDD.scala

```diff
@@ -127,18 +127,12 @@ private object ParallelCollectionRDD {
       })
     }
     seq match {
-      case r: Range.Inclusive => {
-        val sign = if (r.step < 0) {
-          -1
-        } else {
-          1
-        }
-        slice(new Range(
-          r.start, r.end + sign, r.step).asInstanceOf[Seq[T]], numSlices)
-      }
       case r: Range => {
-        positions(r.length, numSlices).map({
-          case (start, end) =>
+        val sign = r.isInclusive && (r.end == Int.MaxValue || r.end == Int.MinValue)
```

Comment:
Try to convert this inclusive range 
``` 
-2 to Int.MinValue by -1
```
to exclusive range will be
```
-2L until -1L + Int.MinValue by -1
```
-1 + Int.MinValue will overflow.

As for sign, which name would you recommend? How about inclusiveRangeWithIntBoundary?